overlord/devicestate/systems_test.go

```diff
@@ -519,26 +544,35 @@ func (s *createSystemSuite) TestCreateSystemInfoAndAssertsChecks(c *C) {
 		info, present := infos[name]
 		return info, present, nil
 	}
+	var observerCalls int
+	snapWriteObserver := func(dir, where string) error {
+		c.Fatalf("unexpected call")
+		observerCalls++
```

Comment:
Nitpick, tracking observerCalls has no effect with Fatalf above it, no? Maybe reorder it or drop?